ebean-agent/src/main/java/io/ebean/enhance/ant/OfflineFileTransform.java

```diff
@@ -142,15 +137,20 @@ private void processPackage(String dir) {
   }
 
   private void transformFile(File file) throws IOException, IllegalClassFormatException {
-
     String className = getClassName(file);
-
-    byte[] result = inputStreamTransform.transform(className, file);
-
-    if (result != null) {
-      InputStreamTransform.writeBytes(result, file);
-      if (listener != null && logLevel > 0) {
-        listener.logEvent("Enhanced " + file);
+    try {
+      byte[] result = inputStreamTransform.transform(className, file);
+      if (result != null) {
+        InputStreamTransform.writeBytes(result, file);
+        if (listener != null && logLevel > 0) {
+          listener.logEvent("Enhanced " + file);
+        }
+      }
+    } catch (EnhancementException e) {
+      if (listener != null) {
+        listener.logError("Error enhancing class " + className + " " + e.getMessage());
```

Comment:
the maven plugin registers a listener. This is the mechanism such that the maven plugin knows this EnhancementException was thrown ... and that means the maven plugin should throw a MojoExecutionException such that the maven build fails.